Merge release-4.8 into development

vadi2 opened the pull request:
<!-- Keep the title short & concise so your mom can understand it -->
#### Brief overview of PR changes/additions
Merge release-4.8 into development
#### Motivation for adding to Mudlet
So nightly builds are 4.8
#### Other info (issues closed, discussion etc)

vadi2 committed:
Fix unreadable text by default in the background button (#3691)

* Fix unreadable text by default in the background button

* Add buddies

* Swap buttons back

(cherry picked from commit a99ea811c06ca668bc26d0859326cedba65376ce)

github-actions[bot] committed:
(autocommit) Updated IRE mapping script to latest upstream (#3695)

Co-authored-by: mudlet-machine-account <mudlet-machine-account@users.noreply.github.com>
(cherry picked from commit 0e0dc43cfa42249069b8cddab5037b4d4c263c6e)

Edru2 committed:
Localization for Lua (#3653)

* first commit

Allow localization of Adjustable Containers right click Menu.

* added translations

translations has the default translations as json file
translated has the finished translation for them as json files

if file in translated cannot be found the default file will be loaded.
The default file can be used also in crowdin as default file.
with the settings /translated/%locale_with_underscore%_%original_file_name% in the file settings

* Format & sort file

* delete translations in lua and put the to the standard path

* change to chrome.json format and put everything it's own folder

Also added the russion translation vadi provided
(plz check if there is no copy/past error ;D)

* lang code at the end and less redundant foldername

* find directory approach

* Update Other.lua

* add folder to check for appimage

* Build using patched version of installers

* another try

* change to only one file for all lua translations

* description also possible now with one file

* clean up and error messages added

* Update src/mudlet-lua/lua/Other.lua

Co-Authored-By: Vadim Peretokin <vperetokin@gmail.com>

* attempt to add translation files to macx

* another go

* Update mudlet.pro

* Update mudlet.pro

* Revert "Build using patched version of installers"

This reverts commit 602cdea0960358a1ca5cd2fff20f128e0651f38c.

* Update mudlet-lua.json

* loadTranslations rewrite to check default

Had to rewrite loadTranslations, so it checks also in default files if it didn't find a translation in translated json

Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>
(cherry picked from commit e6514341e6b3edb3cdd18be26b25fe8b370e4535)

# Conflicts:
#	src/mudlet-lua/lua/geyser/GeyserAdjustableContainer.lua

mudlet-machine-account committed:
New Crowdin translations (#3671)

* New translations mudlet.ts (German)

* New translations mudlet.ts (Portuguese, Brazilian)

* New translations mudlet.ts (Portuguese, Brazilian)

* New translations mudlet.ts (Portuguese, Brazilian)

* New translations mudlet.ts (French)

* New translations mudlet.ts (Spanish)

* New translations mudlet.ts (German)

* New translations mudlet.ts (Greek)

* New translations mudlet.ts (Italian)

* New translations mudlet.ts (Dutch)

* New translations mudlet.ts (Polish)

* New translations mudlet.ts (Portuguese)

* New translations mudlet.ts (Russian)

* New translations mudlet.ts (Turkish)

* New translations mudlet.ts (Chinese Simplified)

* New translations mudlet.ts (Chinese Traditional)

* New translations mudlet.ts (Portuguese, Brazilian)

* New translations mudlet.ts (English, United Kingdom)

* New translations mudlet.ts (Pirate English)

* New translations mudlet.ts (Italian)

* New translations mudlet.ts (Russian)

* New translations mudlet.ts (Portuguese, Brazilian)

* New translations mudlet.ts (German)

* New translations mudlet.ts (Russian)

Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>
(cherry picked from commit cef29bca8cb9bb85b14bfc316efd81f8464c98f6)

vadi2 committed:
Add Carrion Fields as a default MUD (#3697)

* Add Carrion Fields as a default MUD

* Rename .org to .net

* Update src/dlgConnectionProfiles.cpp

Co-Authored-By: Kebap <leckerkebap@gmx.li>

* Sort files list

* Remove extra braces

* Update to a higher res icon

Co-authored-by: Kebap <leckerkebap@gmx.li>
(cherry picked from commit 85aa7a6bd05bf91235e00b0c57f88f0914aef521)

imgbot[bot] committed:
[ImgBot] Optimize images (#3710)

/src/icons/carrionfields.png -- 57.09kb -> 35.58kb (37.68%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

Co-authored-by: ImgBotApp <ImgBotHelp@gmail.com>
(cherry picked from commit 61332c789d95aa48644588279b2b62e05fadb361)

vadi2 committed:
4.8.0 release

vadi2 committed:
Install pcregrep for Travis

vadi2 committed:
Back to -dev

gcms committed:
Fixes #3723 - MXP &text; not working as expected (#3724)

* Fixes #3723 - MXP &text; not working as expected

- Fixes problem with interpolating `&text;` in standalone `<SEND>` tags

* Fixes problem with semicolons not being replaced

- Similar problem to the uppercased `&TEXT;`, the custom tag handler stripped the semicolons, but the handler for send tag didn't.

vadi2 committed:
Move latest added MUD to bottom of list (#3741)


(cherry picked from commit fcd5c987a571ff08f760ae8add1e310d71e65df7)

Kebap reviewed (approved):
Can't resolve conflicts, please review

Edru2 committed:
first commit (#3737)

(cherry picked from commit 2a891cf2c3351bbe4a517205c22cb9b1b91dd3c7)

Edru2 committed:
Fix Adjustable.Container customMenu (#3750)

(cherry picked from commit f88eda1eaab9a0b8f75543e01bad718228765c5b)

Edru2 committed:
Restore :new behaviour and add :new2 (#3738)

* first commit

* code form improvement, change name to add2 and inheritance

* original geyser.add, exclude hbox/vbox/adjc overwrite

Noticed that hbox/vbox didn't work with add2 as they have their own add functions and if passed on (inheritance) was active their function were overwritten what made them not functional anymore.
(The same is true for adjustable containers)

Geyser.add is reverted to be the original (no inheritance possible for that anymore, as it's not even needed)

This commit solves this issues.

* new2 and useAdd2 to decide wheter use add2 or the old add

Use new2 as constructor for a container to automatically add2 for it's elements.

if only an element should use add2 then use new2 only for that one.

If a single element in an new2 container should use the old add put useAdd2 = false in that elements constructor

(cherry picked from commit 76db16fffea757e1557331dbba3c4e56fe72dc58)

# Conflicts:
#	src/mudlet-lua/lua/geyser/GeyserAdjustableContainer.lua

vadi2 committed:
4.8.1 bugfix update

vadi2 committed:
4.8.2 update

vadi2 committed:
Merge remote-tracking branch 'origin/development' into release-4.8

# Conflicts:
#	src/TBuffer.cpp
#	src/dlgConnectionProfiles.cpp
#	translations/lua/translated/mudlet-lua_de_DE.json
#	translations/lua/translated/mudlet-lua_it_IT.json
#	translations/lua/translated/mudlet-lua_ru_RU.json
#	translations/translated/mudlet_de_DE.ts
#	translations/translated/mudlet_el_GR.ts
#	translations/translated/mudlet_en_GB.ts
#	translations/translated/mudlet_en_PT.ts
#	translations/translated/mudlet_es_ES.ts
#	translations/translated/mudlet_fr_FR.ts
#	translations/translated/mudlet_it_IT.ts
#	translations/translated/mudlet_nl_NL.ts
#	translations/translated/mudlet_pl_PL.ts
#	translations/translated/mudlet_pt_BR.ts
#	translations/translated/mudlet_pt_PT.ts
#	translations/translated/mudlet_ru_RU.ts
#	translations/translated/mudlet_tr_TR.ts
#	translations/translated/mudlet_zh_CN.ts
#	translations/translated/mudlet_zh_TW.ts

vadi2 committed:
Re-set dev

vadi2 committed:
Upgrade to Qt 5.12.8 for Linux builds (#3781)

* Upgrade to Qt 5.12.8 for Linux builds

* Try not specifying flags

(cherry picked from commit 1a3253505d146604c30aa2efdd207bf7aab5d4c7)
